daq/host.py

```diff
@@ -247,6 +253,12 @@ def _prepare(self):
         self._state_transition(_STATE.WAITING, _STATE.INIT)
         self.record_result('sanity', state=MODE.DONE)
         self.record_result('dhcp', state=MODE.EXEC)
+        _ = list(listener(self) for listener in self._dhcp_listeners)
+
+    def register_dhcp_ready_listener(self, callback):
+        """Registers callback for when the host is ready for activation"""
+        if callback:
```

Comment:
when is callback not defined -- is there ever a valid use-case for that?  Seems like it would be better to error out in that case or pass it along.  Silently ignoring errors usually isn't a good thing.